### PR TITLE
Monitoring Server

### PR DESCRIPTION
**Pull request by piotrek6641**
**What?**  
- Monitoring server package that allows checking health of a process such as STH/ manager
- `ModuleLoader` to load additional modules on demand. Module should be installed manually, for example with: 
`npm i @scramjet/monitoring-server -g` 

**Usage:**
- **Install `@scramjet/monitoring-server` package**.
- Start STH with a flag `--healtz-port 3030`
- make a `GET` request to `/healthz` endpoint (example `curl http://localhost:3030/healtz`)
- Will return `ok` with a status `200` if everything is working well, `500` otherwise.

- Available flags: 
```
  --healtz-port <healtz-port>              Starts monitoring sever on a selected port
  --healtz-host <healtz-host>              Starts monitoring sever on a specified interface e.g ["0.0.0.0"]. Requires --healtz-port
  --healtz-path <healtz-path>              Exposes monitoring endpoint on specified path. Requires --healtz-port
```

**Clickup Task:** 
https://app.clickup.com/t/24308805/VDM-1350

**Review checks:**

These aspects need to be checked by the reviewer:

- [ ] Verify and confirm operation (please post a screenshot) 
- [ ] All STH tests pass
- [ ] All [Scramjet Cloud Platform](https://docs.scramjet.org/platform) tests pass
- [ ] Documentation is updated or no changes

### CODE REVIEW

File: packages/monitoring-server/src/monitoring-server.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        this.options = options;
+    }
+
+    startServer() {
+        const server = createServer(async (req, res) => {
+            if (req.url === "/healtz" && req.method === "GET") {
+                let ok = true;
+
+                if (this.options.validator) {
+                    ok = await this.options.validator();
```
Comment (patuwwy) [inline packages/monitoring-server/src/monitoring-server.ts:16]: try / catch

File: packages/monitoring-server/src/monitoring-server.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    startServer() {
+        const server = createServer(async (req, res) => {
+            if (req.url === "/healtz" && req.method === "GET") {
+                let ok = true;
+
+                if (this.options.validator) {
+                    try {
+                        ok = await this.options.validator();
+                    } catch (_e) {
```
Comment (alicja-gruzdz) [inline packages/monitoring-server/src/monitoring-server.ts:18]: Let's log this error

**Review by alicja-gruzdz — COMMENTED** [review]
thanks, @piotrek6641 :raised_hands: 

Could you please also perform a quick memory profiling to verify how much the mem consumption increased after adding another server? I know it's lightweight, but let's measure it.

**Review by alicja-gruzdz — APPROVED** [review]
excellent

**Comment by piotrek6641** [comment]
### Without montioring server:
image
### With monitoring server:
image

### Calling endpoint:
image

**Review by alicja-gruzdz — APPROVED** [review]

**Review by patuwwy — APPROVED** [review]

**Comment by stokowski** [comment]
lgtm, I would like to to test it on dev env :)

**Review by alicja-gruzdz — COMMENTED** [review]
Verify :heavy_check_mark: 
Screenshot from 2023-09-21 12-36-31

Screenshot from 2023-09-21 12-36-41

**Comment by alicja-gruzdz** [comment]
Ready to merge.

**Pull request merged**